api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1080,6 +1081,18 @@ type ClusterNetworking struct {
 	//
 	// +optional
 	APIServer *APIServerNetworking `json:"apiServer,omitempty"`
+
+	// allocateNodeCidrs is a flag that controls whether the KCM manages node CIDR
+	// allocation. When using networkType=Other, it is recommended to set this field
+	// to "true" if Flannel is used as the CNI, as it relies on this behavior.
+	// Default is false.
+	// This field can only be set to true when NetworkType is "Other". Setting it to true
+	// with any other NetworkType will result in a validation error during cluster creation.
+	//
+	// +optional
+	// +kubebuilder:default:=false
```

Comment:
avoid setting defaults on the API level, default in code instead and mention the current default in the field description